app/services/highlighter/ai-highlighter/ai-highlighter.ts

```diff
@@ -47,3 +48,208 @@ export interface IHighlighterMessage {
   type: EHighlighterMessageTypes;
   json: {};
 }
+interface IHighlighterProgressMessage {
+  progress: number;
+}
+
+// Buffer management class to handle split messages
+class MessageBufferHandler {
+  private buffer: string = '';
+  private readonly startToken: string = '>>>>';
+  private readonly endToken: string = '<<<<';
```

Comment:
i feel like these should probably be constants defined outside of the class but its not a huge deal